### DISCUSSION TITLE
Support for Elide GraphQL enpoints

### DISCUSSION THREAD

**Discussion by chirdeeptomar** [refinedev/refine#1785] [Ideas]
I use an Open Source framework called [Elide] (https://elide.io/pages/guide/v6/01-start.html) for building API's in GraphQL and REST. I am building the front-end using Refine to query the GraphQL server. 

Elide query conventions are different than how the graphql provider currently works, I would like to write custom GraphQL queries but still benefit from Refine's capabilities like sorting, filtering etc. Essentially I would like to override some of the way graphql provider works. 

Sample Query:

```gql
{ 
    query: { group(filter="name=='Java*'", sort="name") { edges { node { name commonName description } } } }
}
```

Output:
```json
 {
    "data": {
        "group": {
            "edges": [
            {
                "node": {
                "commonName": "Example Repository",
                "description": "The code for this project",
                "name": "com.example.repository"
                }
            },
            {
                "node": {
                "commonName": "Elide",
                "description": "The magical library powering this project",
                "name": "com.yahoo.elide"
                }
            }
            ]
        }
    } 
```

Not sure how much work is to write a Elide Data Provider.

**Comment by omeraplak** [comment]
Hey @chirdeeptomar 👋,
Thank you for contacting us. We will deal with it next week 🚀

**Comment by omeraplak** [comment]
Hey @chirdeeptomar 👋,
I was able to build the query you shared with us with [gql-query-builder](https://github.com/atulmy/gql-query-builder).

```ts
        const query = gql.query({
            operation: "group",
            variables: { filter: "name=='Java*'", sort: "name" },
            fields: [
                {
                    edges: [
                        {
                            node: ["name", "commonName", "description"],
                        },
                    ],
                },
            ],
        });
```
output;
```ts
    {
      query: 'query ($filter: String, $sort: String) { group (filter: $filter, sort: $sort) { edges { node { name, commonName, description } } } }',
      variables: { filter: "name=='Java*'", sort: 'name' }
    }
```

If you want to create `filter` and `sorting` dynamically we have many examples. Some that I think might help you,

Hasura (GraphQL): https://github.com/pankod/refine/blob/master/packages/hasura/src/index.ts#L69
Strapi (GraphQL): https://github.com/pankod/refine/blob/master/packages/strapi-graphql/src/index.ts#L19
Strapi v4 (Rest): https://github.com/pankod/refine/blob/master/packages/strapi-v4/src/dataProvider.ts#L58
Airtable (Rest): https://github.com/pankod/refine/blob/master/packages/airtable/src/index.ts#L43
Data Provider Doc: https://refine.dev/docs/core/providers/data-provider/#getlist

I hope it will be useful for you 🚀
